docker/app
# Push local docker images and use relocation map

**- What I did**

- Allow to push built images that are only available inside the local docker image store.
- Remove `platform` and `tag` flags from the `push` command.
- Bump CNAB schema version to **v1.0.0**
- Store and use the relocation map generated by `cnab-to-oci` (see https://github.com/docker/cnab-to-oci/issues/58)

**- How I did it**

Integration of the latest `cnab-to-oci` that performs the push.

_Note for the reviewer:_ the change is not that big, but contains multiple libraries bump so the diff is really big.
Changes are in different commits to be easier to understand and review. Bump commits contains both bump and immediate changes to be able to compile. But e2e tests are updated in dedicated commits.
One e2e test is still deactivated the time I found why it doesn't work well.

**- How to verify it**

- check the help of `docker app push` command: should not contain option anymore
- build an application that contains local images (like `dockercoins` example)
   - `docker app build -t user/repo/dockercoins:push .`
- push it to a registry
   - `docker app push user/repo/dockercoins:push`
- all images must be available in the registry after the push:
  - invocation image (not tagged anymore with `-invoc`)
  - service images that already exists in the registry (like if you reference a `redis`) must be mounted or copied (like before)
  - service images that were built by `docker app build`
- drop all images built (or use an other machine)
   - `for id in $(cat bundle.json | jq '.images | to_entries[] | .value.contentDigest | select(. != null)' | sed 's/"//g' | sed 's/sha256\://g'); do docker rmi -f $id; done`
- pull it from the registry
   - `docker app pull user/repo/dockercoins:push`
   - no dockercoins built images are pulled here, only bundle and relocation map
- run it
   - `docker app run user/repo/dockerocoins:push`
   - images are pulled at runtime

**- Description for the changelog**

`push` command now pushes missing images from the local docker daemon image store to the registry. Options about platform and tags has been removed and must be defined in a previous step, like during build.
CNAB schema is now in version 1.0.0.
